### ISSUE TITLE
Sender keeps to lose connection with RPI when transferring large files (PS4 11.00, macOS 14) 

### ISSUE THREAD

**Issue by serista** [Gkiokan/ps4-remote-pkg-sender#79]
RPS v2 2.10.4
RPI from the Homebrew store.
Mac Mini M1 connected to PS4 with LAN.
Plain FTP transfer works fine for any files.

When installing large games, eventually after about 5-20 minutes RPS stops to update progress bar and estimated time, also pause/resume stops to work. On the PS4 side RPI continues to receive file to the very end, but after the download finished PS4 shows a message that the file can not be installed.

Works fine for smaller files.

**Comment by Gkiokan** [comment]
Hi there,

thanks for reporting back the issue! 

Probably same issue as the Issue https://github.com/Gkiokan/ps4-remote-pkg-sender/issues/78#issuecomment-2133466750

Explanation in the Troubleshooting Guide:
https://github.com/Gkiokan/ps4-remote-pkg-sender/blob/master/Troubleshoot.md#my-rpi-crashes-on-console
```
My RPI crashes on console
This happend to a couple of users when we try to hit RPI too hard.
RPI cann't handle too many concurrent requests and crashes after a while
especially when you try to install something big 100GB+ PKG files or
installing multiple files at once and have a low update interval value.

It may help if you leave the interval value arround 2-3 secs because
updating progress info is also a request that will be sent upon the interval.

In v2.4.2 you can still send as many install request as you want, that is not limited yet,
but consider the fact that once the installation starts, RPI has nothing more to do with it.
Download still continues on PS4 even if you jump out of RPI! Keep that in mind!
```

The Solution to this is first update the Update Interval to max, that will prevent the RPI to crash later then sooner.  
If you want to keep update ticks, you may restart the RPI on the playstation that will fix the "crashed state" of it. 
That is a known bug with the RPI and happens if you go to hard on it - very large files. 

The RPI is good, but it was made for the 5.05 back then. It never was planed to "life" that long. 
But luckly I am working with Sistro in a embeeded RPI within the GoldHEN. 
This should fix all issues and update the dependencies to the latest Standards.

**Comment by serista** [comment]
Well, I'll try to increase Update interval but the thing is that 1) the RPS may пуе stuck even for files much smaller than 100GB in my case, 2) when it gets stuck and I switch to PS4 and wait until the download is finished the PS4 shows message that the file can not be installed somehow. If not #2, it wouldn't be big problem.

Is it safe to switch from the RPI splash screen to XMB to check the progress?

**Comment by serista** [comment]
Ok, I tried again and this time Uncharted 4 was transferred without getting stuck but still it "could not be installed" in the end somehow. Also tried RDR, the transfer kept stopping immediately on the PS4 end with the same message (tried it 3 times). The very same RDR I installed just fine yesterday by transferring it via FTP and then installing with pkg installer. It's strange, it seems something is wrong with some games being installed via RPI.

**Comment by Gkiokan** [comment]
Interesting insights. That has been never reported but it may be also in conjunction with FW 11.00.  
We will keep an eye on it once we are ready with the new RPI Implementation.

**Comment by Shu2Ouma** [comment]
I wanted to comment that I am experiencing the same issue on my PS4 Fat with firmware 10.01, where the connection is lost from the client side (RPI) during large file transfers. However, this issue does not occur on my PS4 Pro with firmware 9.00. The curious thing is that, even though the connection is lost, the transfer of the file that was being sent during the failure completes correctly. The problem is that the transfer of the remaining files in the queue does not continue. On my PS4 Pro, I have installed up to 10 games in the queue, and the connection has never been lost. This leads me to suspect that the hardware power of the PS4 Fat could be an important factor, as it has a more limited processor and overall performance compared to the PS4 Pro.
(Please note, I am using a translator as I do not speak the language).

**Comment by Gkiokan** [comment]
> I wanted to comment that I am experiencing the same issue on my PS4 Fat with firmware 10.01, where the connection is lost from the client side (RPI) during large file transfers. However, this issue does not occur on my PS4 Pro with firmware 9.00. The curious thing is that, even though the connection is lost, the transfer of the file that was being sent during the failure completes correctly. The problem is that the transfer of the remaining files in the queue does not continue. On my PS4 Pro, I have installed up to 10 games in the queue, and the connection has never been lost. This leads me to suspect that the hardware power of the PS4 Fat could be an important factor, as it has a more limited processor and overall performance compared to the PS4 Pro. (Please note, I am using a translator as I do not speak the language).

Ye that could be. The RPI stops working when it's kinda full - in simple terms. In the other Hand we have still the working Downloads which are unrelated to the RPI. The RPI helps us just to start all this and keep track of the progress % etc. But once it stops, we loose the state unfortunatelly. In most cases RPI restart helps to get back on track but besides that it's an ugly workaround. 

I know we have had a fully working submit for FW until 9.00.  A couple errors here and there has been grown up with 11.xx+ and I am not sure if anyone is gonna update the RPI itself for a better handling as it gets older and older. 

Thanks for your commit and thoughts. We will figure this out somehow, when we get time for it.

**Comment by Zidar05** [comment]
> Interesting insights. That has been never reported but it may be also in conjunction with FW 11.00. We will keep an eye on it once we are ready with the new RPI Implementation.

Is there any progress in this direction? Will there be an update this month? I have fw11.00 slim and this error also appeared recently.. It wasn 't there before .

**Comment by Gkiokan** [comment]
> > Interesting insights. That has been never reported but it may be also in conjunction with FW 11.00. We will keep an eye on it once we are ready with the new RPI Implementation.
> 
> Is there any progress in this direction? Will there be an update this month? I have fw11.00 slim and this error also appeared recently.. It wasn 't there before .

Yes kinda, but the progress is very slow currently. Sistro is working on a GoldHEN Plugin Integration that should work way more stable as RPI and fix all this issues. We traced down the Issues sofar but we both are also heavy on duty right now for the end year business jobs and that takes so much time :(

I hope we can make a big Update Release within Q4.

**Comment by serista** [comment]
I should add that my cable connection is 100Mbit/s, if it's important.

**Comment by Gkiokan** [comment]
@serista  user skyneli found a solution to this. Can you try if that helps you? 
Block the internet access for the ps4 on your router.  
Maybe the same issue on FW11+ is related to your FW10+, too.

https://github.com/Gkiokan/ps4-remote-pkg-sender/issues/95#issuecomment-3429980331

**Comment by serista** [comment]
@Gkiokan, access to account.np.ac.playstation.net has been blocked on my router for ages, my PS4 can't access PSN that's for sure. Just in case, my router and computer connected via wi-fi, not wired, although I don't think it has any relation, as transfer between computer and PS4 goes directly, without router. I'll need to try it again, now that I upgraded to Tahoe and M4, long time didn't touch the RPI.

**Comment by Gkiokan** [comment]
Wi-Fi may be slower but not unstable compared to cable.

As you mentioned, and it's already working for anyone else up to FW9.00 without any problems and most of the users also seem to have a working connection on FW10+, too. But there is this one user that gets it resolved the issue by blocking the complete Internet Access.  

It's worth mentioning that even it makes no sense, the connection issue has been resolved for him.  

FYI: MacOS 14.4+ has (still) probably a permission issue for the Server code in my app, a Patch and PoC is ready, there is a update coming soon. Just in case you wanna touch and try it again on mac and fail to get a Heartbeat from my app on mac.
